examples/function_tools/README.md

```diff
@@ -0,0 +1,37 @@
+# Functions calling
+
+## Deploy a new function
+To avoid repeatedly specifying all attributes related to tools and tool_choice when using functions calling, you can create a classe inheriting from ```functions.ToolFunctions```.
+
+You need to instantiate 4 attributes: 
+ - description (string)
+ - parameters (dict)
+ - name (string)
+ - tool_type (string) 
+
+Each attributes correpondings to [Openai api](https://platform.openai.com/docs/api-reference/chat/create#chat-create-tools)
+
+After the class is created, you have to declare in TOOLS_DICT and TOOLS global variables (add weather function tool for example).
+
+```
+TOOLS_DICT = {
+    'weather': Weather(),
+}
+TOOLS = ['weather']
+```
+
+## Called functions
```

Comment:
Should we specify that this endpoints has to be called, only if happy_vllm server is deployed with pre defined tools?